components/script/dom/htmlcanvaselement.rs

```diff
@@ -279,7 +279,7 @@ impl HTMLCanvasElementMethods for HTMLCanvasElement {
             let encoded = encoded.to_base64(STANDARD);
             Ok(DOMString::from(format!("data:{};base64,{}", mime_type, encoded)))
         } else {
-            Err(Error::NotSupported)
+            Ok(DOMString::from(format!("data:image/png;")))
```

Comment:
"When its canvas context mode is none, a canvas element has no rendering context, and its bitmap must be fully transparent black with an intrinsic width equal to the numeric value of the element's width attribute and an intrinsic height equal to the numeric value of the element's height attribute, those values being interpreted in CSS pixels, and being updated as the attributes are set, changed, or removed." (https://html.spec.whatwg.org/multipage/scripting.html#the-canvas-element).

In other words, a canvas always has content, so we must always return a valid image with the specified width and height.